src/inmanta/data/dataview.py

```diff
@@ -1211,16 +1213,19 @@ def get_base_url(self) -> str:
 
     def get_base_query(self) -> SimpleQueryBuilder:
         base = SimpleQueryBuilder(
-            select_clause="""SELECT a.name, a.environment, last_failover, paused, unpause_on_resume,
-                                            ap.hostname as process_name, ai.process as process_id,
-                                            (CASE WHEN paused THEN 'paused'
-                                                WHEN id_primary IS NOT NULL THEN 'up'
-                                                ELSE 'down'
+            select_clause="""SELECT a.name, a.environment, a.last_failover, a.paused, a.unpause_on_resume,
+                sa_join.hostname as process_name, sa_join.process as process_id,
+                                            (CASE WHEN a.paused THEN 'paused'
+                                                WHEN NOT a.paused AND sa_join.id_primary IS NULL THEN 'down'
+                                                ELSE 'up'
                                             END) as status""",
-            from_clause=f" FROM {Agent.table_name()} as a LEFT JOIN public.agentinstance ai ON a.id_primary=ai.id "
-            " LEFT JOIN public.agentprocess ap ON ai.process = ap.sid",
-            filter_statements=[" a.environment = $1 "],
-            values=[self.environment.id],
+            from_clause=f" FROM {Agent.table_name()} a "
+            "LEFT JOIN (SELECT sa.name, sa.id_primary, ap.hostname, ai.process FROM agent sa  "
+            f"LEFT JOIN {AgentInstance.table_name()} ai ON sa.id_primary=ai.id "
+            f"LEFT JOIN {AgentProcess.table_name()} ap ON ai.process = ap.sid  "
+            "WHERE sa.environment = $1 AND sa.name = $2) sa_join ON a.name <> sa_join.name ",
```

Comment:
the join on `<>` I don't know about, my SQL is too weak. 
@Hugo-Inmanta can you check this out?

Reply:
I think it is working as expected i.e. for every nominal agent we retrieve the info for the hostname and pid of the scheduler in the env.
@hlloreda  do I understand correctly that this `ON` clause is there just to have something to join on ? ie we would expect the same result if we passed `ON true` ? In this case a `CROSS JOIN` would be preferable I think